src/main/java/ai/datahunters/md/server/photos/upload/ArchiveHandler.java

```diff
@@ -0,0 +1,89 @@
+package ai.datahunters.md.server.photos.upload;
+
+import org.apache.commons.compress.archivers.ArchiveEntry;
+import org.apache.commons.compress.archivers.ArchiveException;
+import org.apache.commons.compress.archivers.ArchiveInputStream;
+import org.apache.commons.compress.archivers.ArchiveStreamFactory;
+import org.apache.commons.compress.archivers.dump.UnrecognizedFormatException;
+import org.apache.commons.compress.compressors.CompressorException;
+import org.apache.commons.compress.compressors.CompressorStreamFactory;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.tika.Tika;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class ArchiveHandler {
+    public static final String GTAR = "application/x-gtar";
+    public static final String TAR = "application/x-tar";
+    public static final String ZIP = "application/zip";
+    public static final String GZ = "application/gzip";
+    public static final String XZ = "application/x-xz";
+    public static final String BZ2 = "application/x-bzip2";
+
+    @Value("${dirForUploadedFiles}")
```

Comment:
would it be possible to get this guy in constructor and to autowire put the new instance somewhere? And would be nice to make it accept `Path` instead of string

Reply:
When you have this constructor which accept the dir and the `FileService` it should be quite straight forward to test